src/app/admin/createUser/UserRoleCard.tsx

```diff
@@ -3,14 +3,15 @@ import React from "react";
 import { getDropdownListHandler } from "@/components/DataInput/inputHandlerFactories";
 import UserRoleDropdownInput from "@/app/admin/common/UserRoleDropdownInput";
 import { InviteUserCardProps } from "./CreateUserForm";
+import { UserRole } from "@/databaseUtils";
 
 const UserRoleCard: React.FC<InviteUserCardProps> = ({ fieldSetter }) => {
     return (
         <GenericFormCard title="User Role" required>
             <UserRoleDropdownInput
-                defaultValue="caller"
+                defaultValue="volunteer"
                 onChange={getDropdownListHandler((role: string) =>
-                    fieldSetter([{ key: "role", value: role }])
+                    fieldSetter({ role: role as UserRole })
```

Comment:
instead of asserting the value as UserRole here, can we make it so that `UserRoleDropdownInput` has a prop onChange that is `(userRole: UserRole) => void` or something similar?

Reply:
I gave this a go yesterday but I gave up because I got this error and I didn't understand it :'(
![image](https://github.com/NorwoodAndBrixtonFoodbank/nbf-website/assets/138508884/ba1f43f9-0dab-46b0-91a4-0636bc7c5a4a)
Please help when you have some time :)

Reply:
I think you meant to use `Value extends string`.

Regardless you'll see an error message.
```
Argument of type string | Value is not assignable to parameter of type Value
string | Value is assignable to the constraint of type Value, but Value could be instantiated with a different subtype of constraint string
```


You'd probably have to delegate the validation of the value to the function caller i.e.
```ts
export const getDropdownListHandler = <Value extends string>(
    setValue: ValueSetter<Value>,
    isValueValid: (value: string) => value is Value
): SelectChangeEventHandler<Value> => {
    return (event: SelectChangeEvent<Value>) => {
        const value = event.target.value;
        if (isValueValid(value)) {
            setValue(value);
        }
    };
};
```

Reply:
Discussed in person and think is fixed